cli/crates/server/src/bridge/server.rs

```diff
@@ -121,14 +138,98 @@ async fn search_endpoint(
     Ok(Json(response))
 }
 
+#[allow(clippy::too_many_lines)]
 async fn invoke_resolver_endpoint(
     State(handler_state): State<Arc<HandlerState>>,
     Json(payload): Json<ResolverInvocation>,
 ) -> Result<Json<serde_json::Value>, ApiError> {
     trace!("resolver invocation\n\n{:#?}\n", payload);
+
+    let environment = Environment::get();
+
+    let resolver_worker_port = loop {
+        let mut resolver_builds = handler_state.resolver_builds.lock().await;
+
+        let notify = if let Some(resolver_build) = resolver_builds.get(&payload.resolver_name) {
+            match resolver_build {
+                ResolverBuild::Succeeded { worker_port, .. } => break *worker_port,
+                ResolverBuild::Failed => return Err(ApiError::ResolverSpawnError),
+                ResolverBuild::InProgress { notify } => {
+                    if Arc::strong_count(notify) == 1 {
```

Comment:
What's the purpose here of this branch? Shouldn't you always wait where it's `InProgress`?

Reply:
It's to handle interrupted builds when a request invoking a resolver gets cancelled by the client. I agree this isn't a great way to handle this, but for a better approach we would need a dedicated resolver build server which I'd like to do in a future PR when I do the single HTTP server refactoring.

Reply:
Ah got it, didn't realize at first `Arc::strong_count(notify) == 1` meant that the `resolver_builds` was the sole owner of notify. Might be worth a comment IMHO.

Reply:
Added, a comment. 🙂

Reply:
Thanks!